crates/ty_python_semantic/src/types.rs

```diff
@@ -7614,6 +7740,10 @@ pub enum KnownInstanceType<'db> {
     /// A constraint set, which is exposed in mdtests as an instance of
     /// `ty_extensions.ConstraintSet`.
     ConstraintSet(TrackedConstraintSet<'db>),
+
+    /// An identity function created with `typing.NewType(name, base)`, which behaves like a
```

Comment:
nit: when you call it, it has an identity-function signature, but it is not actually an instance of `types.FunctionType` at runtime (it's an instance of `typing.NewType`)

```suggestion
    /// An identity callable created with `typing.NewType(name, base)`, which behaves like a
```